### PR TITLE
feat: update react-scripts

### PR DESCRIPTION
**Pull request by ghost**
See: https://dhis2.atlassian.net/browse/CLI-73.

After the upgrade I was running into this issue: https://github.com/facebook/create-react-app/issues/12177. I've rebuilt the lockfile, that has addressed it. It does mean that everything has been updated to latest though (within the package.json version ranges). If all updated packages properly followed semver that should be fine, but we should double check to make sure. There might also be less drastic ways of addressing the issue.

I've disabled the import extensions rule for now. Besides that I've clarified the logic in the shell eslint config to what made it a little easier to read for me. In that config I also removed the react plugin for the local linting, because that plugin is already present in the cli style config, which caused eslint to fail with a warning.

Changelog for the move from 4 to 5: https://github.com/facebook/create-react-app/blob/main/CHANGELOG.md#500-2021-12-14

### CODE REVIEW

**Review by Mohammer5 — COMMENTED** [review]
> I've disabled the import extensions rule for now

What was the reason for this?

File: shell/.eslintrc.js [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 
-const delegateEslintConfig = path.resolve(__dirname, '../.eslintrc.js') // This should only exist when in development!
-const shouldDelegate =
-    !eslintRunningLocally && fs.existsSync(delegateEslintConfig)
+const isRunningHere = process.cwd() === __dirname
+const hasRootConfig = fs.existsSync(rootConfigPath)
+const isDevelopment = !isRunningHere && hasRootConfig
+
+// Use local config for developing this library, react-app preset for linting app code
+const extendsList = isDevelopment ? rootConfigPath : 'react-app'
```
[the remark this answers is not in the document]
Comment (ismay) [reply shell/.eslintrc.js:12]: That would be cleaner yeah, but it's a reserved word. So I kept the original name. We could rename it to something else though.

**Comment by ismay** [comment]
> > I've disabled the import extensions rule for now
> 
> What was the reason for this?

For speed. It would be a lot of work to address this. I could, but I'd have to postpone a couple other tasks (again).

File: shell/.eslintrc.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mohammer5) [reply shell/.eslintrc.js:12]: Ah, makes sense that it's a reserved word. How about:

```js
module.exports = {
  // Use local config for developing this library, react-app preset for linting app code
  ignorePatterns: isDevelopment ? [] : ['src/D2App/*'],
  // Use local config for developing this library, react-app preset for linting app code
  extends: isDevelopment ? rootConfigPath : 'react-app',
}
```

**Review by Mohammer5 — APPROVED** [review]
> For speed. It would be a lot of work to address this. I could, but I'd have to postpone a couple other tasks (again).

I don't think I'm the right person to judge whether this is the right call or not. I'll approve, is there a way to address adding back the plugin? And if not, how do we document this / prioritize this?

**Comment by ismay** [comment]
> > For speed. It would be a lot of work to address this. I could, but I'd have to postpone a couple other tasks (again).
> 
> I don't think I'm the right person to judge whether this is the right call or not. I'll approve, is there a way to address adding back the plugin? And if not, how do we document this / prioritize this?

I think there's a misunderstanding. So the plugin I removed (eslint react preset) was removed because it was duplicated. Effectively it's still there, just not twice like before. So we don't have to add that back.

The import rule is just disabled. Eventually it should be enabled and the violations should be addressed. Currently it doesn't really affect anything though. The import extensions rule is there for compliance with the es module spec that does not allow omitting the extension. This is not currently a thing for our tooling.

**Review by jenniferarnesen — APPROVED** [review]

File: .eslintrc.js [hunk 2]
```diff
@@ -5,6 +5,9 @@ module.exports = {
     parserOptions: {
         allowImportExportEverywhere: true,
     },
+    rules: {
+        'import/extensions': 'off',
+    },
```
Comment (HendrikThePendric) [inline .eslintrc.js:10]: Was wondering why we need that. Typescript?

File: shell/.eslintrc.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ismay) [reply shell/.eslintrc.js]: Yeah that looks simpler, changed it in [86bd3f9](https://github.com/dhis2/app-platform/pull/721/commits/86bd3f999007b0abb9357936199f61a034ae8d46).

File: .eslintrc.js (hunk 2 above)
Comment (ismay) [reply .eslintrc.js:10]: https://github.com/dhis2/app-platform/pull/721#issuecomment-1160497608

I can take care of it after we've dealt with the functional part of these changes. Having it not block these changes makes it a little easier for me to integrate it with other tasks.

**Review by HendrikThePendric — APPROVED** [review]
LGTM. Releasing a beta first and seems sensible. Could you also capture the follow-up work reg. the linting in a separate issue? _(And post a link to this issue in this PR)_

**Pull request merged**